code/modules/projectiles/guns/energy.dm

```diff
@@ -103,8 +123,12 @@
 		var/iconK = "knife"
 		overlays += image(icon = 'icons/obj/guns/attachments.dmi', icon_state = iconK, pixel_x = knife_x_offset, pixel_y = knife_y_offset)
 	if(itemState)
-		itemState += "[ratio]"
-		item_state = itemState
+		if(multistate == 1)
```

Comment:
What happens if multistate is not 1 and shaded_charge isn't null? Should there be a default case here?